.github/workflows/update-docs.yml

```diff
@@ -0,0 +1,53 @@
+name: Update Template Documentation
+
+on:
+  schedule:
+    - cron: '0 16 * * 2' # Every Tuesday at 11 AM EST (16:00 UTC)
+  workflow_dispatch:
+
+jobs:
+  update-docs:
+    name: Update Template Documentation
+    runs-on: ubuntu-latest
+    if: github.repository == 'GoogleCloudPlatform/DataflowTemplates'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Install Templates Plugin
+        run: |
+          mvn clean install -pl plugins/templates-maven-plugin -am
+      - name: Generate Documentation
+        run: |
+          mvn clean prepare-package \
+            -DskipTests \
+            -PtemplatesSpec
+
+      - name: Check for changes
+        id: changes
+        run: |
+          git status --porcelain
+          if [[ -z $(git status --porcelain) ]]; then
+            echo "no_changes=true" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Create Pull Request
+        if: steps.changes.outputs.no_changes != 'true'
+        uses: peter-evans/create-pull-request@v7
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
We should grant the token write permissions explicitly - https://github.com/peter-evans/create-pull-request?tab=readme-ov-file#token - not sure if this will happen by default or not, but it is something that could be changed in the future at an org level I think